heudiconv/dicoms.py

```diff
@@ -314,19 +315,55 @@ def group_dicoms_into_seqinfos(files, grouping, file_filter=None,
     return seqinfos
 
 
-def get_dicom_series_time(dicom_list):
-    """Get time in seconds since epoch from dicom series date and time
-    Primarily to be used for reproducible time stamping
+def get_timestamp_from_series(dicom_list: list[str]) -> int:
+    """try to return a timestamp indicating when the first dicom in dicom_list was created, or if that
+    info isn't present then a reproducible integer. This is used in setting mtimes reproducibly
+
+    Args:
+        dicom_list (list[str]): list of strings pointing to existing dicom files
+
+    Returns:
+        int: either an int representing when the first dicom was created, 
+            in number of seconds since epoch, or if no datetime info is found then a hash of the
+            SeriesInstanceUID (meaningless value, but reproducible)
```

Comment:
if a "hash" -- it would not be properly sortable between different dicom series.  I can't recall ATM if that might not make it a trouble somewhere - needs checking on how datetime is used -- if not used anywhere for sorting.